arch/arm/boot/dts/imx6ul-ts7180.dts

```diff
@@ -0,0 +1,1020 @@
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi3>;
+	status = "okay";
+
+	spidevfpga: spi@0 {
+		compatible = "spidev";
+		reg = <0>;
+		spi-max-frequency = <1000000>;
+	};
+
+	spioffbd: spi@1 {
+		compatible = "spidev";
+		reg = <1>;
+		spi-cs-high;
+		spi-max-frequency = <1000000>;
+	};
+
+	/* This is actually the FRAM, which is compatible with the AT25 SPI EEPROM */
+	spifram: spi@2 {
+		compatible = "atmel,at25", "cypress,fm25l16b";
+		reg = <2>;
+		spi-max-frequency = <20000000>;
+		size = <0x800>;
+		address-width = <16>;
+		pagesize = <64>;
+		status = "okay";
+	};
+};
+
+&ecspi4 {
+	fsl,spi-num-chipselects = <1>;
+	cs-gpios = <&gpio4 9 0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi4>;
+	status = "okay";
+
+	wilc_spi@0 {
+		compatible = "microchip,wilc1000", "microchip,wilc3000";
+		reg = <0>;
+		spi-max-frequency = <48000000>;
+		reset-gpios = <&gpio6 0 0>;     /* RESET,  GPIO 160 */
+		chip_en-gpios = <&gpio6 1 0>;   /* ENABLE, GPIO 161 */
+		interrupt-parent = <&gpio5>;    /* IRQ,    GPIO 136 */
+		interrupts = <8 0>;
+		status = "okay";
+	 };
+};
+
+&fec1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet1>;
+	phy-mode = "rmii";
+	phy-handle = <&ethphy0>;
+	phy-supply = <&reg_3v3>;
+	status = "okay";
+};
+
+&fec2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet2>;
+	phy-mode = "rmii";
+	phy-handle = <&ethphy1>;
+	phy-supply = <&reg_3v3>;
+	status = "okay";
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ethphy0: ethernet-phy@2 {
+			clocks = <&clks IMX6UL_CLK_ENET_REF>;
+			clock-names = "rmii-ref";
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <2>;
+		};
+
+		ethphy1: ethernet-phy@1 {
+			clocks = <&clks IMX6UL_CLK_ENET_REF>;
+			clock-names = "rmii-ref";
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <1>;
+		};
+	};
+};
+
+&gpio1 {
+	gpio-line-names =
+		"BOOT_MODE_0",
+		"NA",
+		"I2C_1_CLK",
+		"I2C_1_DAT",
+		"ADC_1",
+		"ADC_2",
+		"ETH_MDIO",
+		"ETH_MDC",
+		"ADC_3",
+		"ADC_4",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"CONSOLE_TXD",
+		"CONSOLE_RXD",
+		"SPARE_1",
+		"EN_485",
+		"UART2_TXD",
+		"UART2_RXD",
+		"CAN_2_TXD",
+		"CAN2_RXD_3V",
+		"UART3_TXD",
+		"UART3_RXD",
+		"UART3_CTS#",
+		"UART3_RTS#",
+		"UART4_TXD",
+		"UART4_RXD",
+		"UART5_TXD",
+		"UART5_RXD";
+};
+
+&gpio2 {
+	gpio-line-names =
+		"ENET1_RX_DATA0",
+		"ENET1_RX_DATA1",
+		"ENET1_RX_EN",
+		"ENET1_TX_DATA0",
+		"ENET1_TX_DATA1",
+		"ENET1_TX_EN",
+		"ENET1_TX_CLK",
+		"ENET1_RX_ER",
+		"ENET2_RX_DATA0",
+		"ENET2_RX_DATA1",
+		"ENET2_RX_EN",
+		"ENET2_TX_DATA0",
+		"ENET2_TX_DATA1",
+		"ENET2_TX_EN",
+		"ENET2_TX_CLK",
+		"ENET2_RX_ER",
+		"SD_CMD",
+		"SD_CLK",
+		"SD_D0",
+		"SD_D1",
+		"SD_D2",
+		"SD_D3",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"NA",
+		"NA";
+};
+
+&gpio3 {
+	gpio-line-names =
+		"HD12_SPI_CS",
+		"JTAG_FPGA_TCK",
+		"JTAG_FPGA_TMS",
+		"JTAG_FPGA_TDI",
+		"WDOG#",
```

Comment:
Can this even be used as a GPIO pin?

Be sure all of these GPIO entries only call out pins that can be used as a GPIO. Regulators, peripherals (CAN, I2C, UART, etc.), etc., all label the pins that are used if they use external pins. These show up in a different column in the `/sys/kernel/debug/gpio` file than the GPIO names and make an important distinction in their use.

On platforms like the TS-7180 where there is no reasonable way to make a UART pin an external GPIO, don't give it a name.